src/Foundation/NSDate.cs

```diff
@@ -42,10 +42,19 @@ public partial class NSDate {
 		public static explicit operator DateTime (NSDate d)
 		{
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if ((secs < -63114076800) || (secs > 252423993599))
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {secs}");
```

Comment:
it's nice to add the value in the exception
but it should be done with the right constructor
https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=netframework-4.8#System_ArgumentOutOfRangeException__ctor_System_String_System_Object_System_String_

Reply:
```suggestion
				throw new ArgumentOutOfRangeException (nameof(d), d, $"{nameof(d)} is outside the range of NSDate {secs} seconds");
```